okio-files/src/commonTest/kotlin/okio/files/FileSystemTest.kt

```diff
@@ -286,6 +295,52 @@ abstract class FileSystemTest(
     }
   }
 
+  @Test
+  fun `file metadata`() {
+    if (!isJvm()) return
+
+    val minTime = clock.now().floorSeconds()
+    val path = base / "file-metadata"
+    path.writeUtf8("hello, world!")
+    val maxTime = clock.now().plus(1.seconds).floorSeconds()
+
+    val metadata = filesystem.metadata(path)
+    assertTrue(metadata.isRegularFile)
+    assertFalse(metadata.isDirectory)
+    assertEquals(13, metadata.size)
+    assertTrue { (metadata.createdAt ?: minTime) in minTime..maxTime }
```

Comment:
Do we expect these to ever be null under test?

Reply:
Mostly depends on the operating system / file system combo. I’d like the tests to reflect the true dynamic range of the values, even if our test coverage rarely runs on say, FAT filesystems.